egs/librispeech/asrtts/run.sh

```diff
@@ -0,0 +1,544 @@
+    compute-cmvn-stats scp:data/${train_set}/feats.scp data/${train_set}/cmvn.ark
+    # dump features for training
+    dump.sh --cmd "$train_cmd" --nj ${nj} --do_delta false \
+        data/${train_set}/feats.scp data/${train_set}/cmvn.ark exp/dump_feats/train ${feat_tr_dir}
+    dump.sh --cmd "$train_cmd" --nj ${nj} --do_delta false \
+        data/${train_paired_set}/feats.scp data/${train_set}/cmvn.ark exp/dump_feats/train_p ${feat_tr_p_dir}
+    dump.sh --cmd "$train_cmd" --nj ${nj} --do_delta false \
+        data/${train_unpaired_set}/feats.scp data/${train_set}/cmvn.ark exp/dump_feats/train_up ${feat_tr_up_dir}
+    dump.sh --cmd "$train_cmd" --nj ${nj} --do_delta false \
+        data/${dev_set}/feats.scp data/${train_set}/cmvn.ark exp/dump_feats/dev ${feat_dt_dir}
+    dump.sh --cmd "$train_cmd" --nj ${nj} --do_delta false \
+        data/${eval_set}/feats.scp data/${train_set}/cmvn.ark exp/dump_feats/eval ${feat_ev_dir}
+fi
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    echo "stage 1: JSON preparation for TTS and ASR"
+    echo "dictionary: ${dict}"
+    ### Task dependent. You have to check non-linguistic symbols used in the corpus.
+    mkdir -p data/lang_char/
+    echo "<unk> 1" > ${dict} # <unk> must be 1, 0 will be used for "blank" in CTC
+    if [ $use_bpe == 'true' ]; then
+        cut -f 2- -d" " data/${train_set}/text > data/lang_char/input.txt
+        spm_train --input=data/lang_char/input.txt --vocab_size=${nbpe} --model_type=${bpemode} --model_prefix=${bpemodel} --input_sentence_size=100000000
+        spm_encode --model=${bpemodel}.model --output_format=piece < data/lang_char/input.txt | tr ' ' '\n' | sort | uniq | awk '{print $0 " " NR+1}' >> ${dict}
+    else
+        echo "make a non-linguistic symbol list"
+        #cut -f 2- data/${train_set}/text | tr " " "\n" | sort | uniq | grep "<" > ${nlsyms}
+        #cat ${nlsyms}
+        text2token.py -s 1 -n 1 data/${train_set}/text | cut -f 2- -d" " | tr " " "\n" \
+        | sort | uniq | grep -v -e '^\s*$' | awk '{print $0 " " NR+1}' >> ${dict}
+    fi
+    wc -l ${dict}
+    # make json labels
+    if [ ! -s ${feat_ev_dir}/data.json ]; then
+    data2json.sh --feat ${feat_tr_dir}/feats.scp \
+         data/${train_set} ${dict} > ${feat_tr_dir}/data.json
+    data2json.sh --feat ${feat_tr_p_dir}/feats.scp \
+         data/${train_paired_set} ${dict} > ${feat_tr_p_dir}/data.json
+    data2json.sh --feat ${feat_tr_up_dir}/feats.scp \
+         data/${train_unpaired_set} ${dict} > ${feat_tr_up_dir}/data.json
+    data2json.sh --feat ${feat_dt_dir}/feats.scp \
+         data/${dev_set} ${dict} > ${feat_dt_dir}/data.json
+    data2json.sh --feat ${feat_ev_dir}/feats.scp \
+         data/${eval_set} ${dict} > ${feat_ev_dir}/data.json
+    fi
+    # Make MFCCs and compute the energy-based VAD for each dataset
+    mfccdir=mfcc
+    vaddir=mfcc
+    for name in ${train_set} ${dev_set} ${eval_set}; do
+        if [ ! -s data/${name}_mfcc/feats.scp ]; then
```

Comment:
I think feats.scp is made by make_mfcc.sh.
This sentence will be skipped at re-running if the problem happen after make_mfcc.sh.
Is this all right?